### ISSUE TITLE
Add a `.log` file viewer which correctly formats ANSI escape codes

### ISSUE THREAD

**Issue by evansd** [opensafely-core/airlock#664]
Various OpenSAFELY images (in particular, but not exclusively `ehrql`) write colourised/formatted output to stderr using ANSI escape codes. For example:

![Screenshot from 2023-08-18 10-57-00](https://github.com/opensafely-core/ehrql/assets/19630/16610165-da23-4df1-8d3e-22c324ce6d63)

If we render this in a naive text viewer we get stuff like:
```
[2m2024-09-10 11:35:21[0m [[32m[1minfo     [0m] [1mCompiling dataset definition from demo_dataset.py[0m
[2m2024-09-10 11:35:22[0m [[32m[1minfo     [0m] [1mGenerating dummy dataset[0m
[2m2024-09-10 11:35:22[0m [[32m[1minfo     [0m] [1mAttempting to generate 10 matching patients (random seed: BwRV3spP, timeout: 60s)[0m
[2m2024-09-10 11:35:22[0m [[32m[1minfo     [0m] [1mUse `dataset.configure_dummy_data(population_size=N)` to change number of patients generated[0m
[2m2024-09-10 11:35:23[0m [[32m[1minfo     [0m] [1mBuilding dataset and writing results[0m
```

It would be great if we could render the escape codes into appropriate HTML so it all looks nice. I very much hope there are non-awful libraries which would handle this for us, but I haven't actually looked yet.

**Comment by evansd** [comment]
As a very first pass, just stripping the escape codes out would be an improvement.

**Comment by evansd** [comment]
There's this library, which might be acceptable:
https://github.com/pycontribs/ansi2html

I also see that `nbconvert` has a standalone module for doing this which we could vendor:
https://github.com/jupyter/nbconvert/blob/e1599627fd1f9e06d983e3974eadb58d2a92c44c/nbconvert/filters/ansi.py

**Comment by rebkwok** [comment]
Closed by #673
